### PR TITLE
feat: user can index with their own data

### PR DESCRIPTION
**Pull request by Hippopotamus0308**
index user's own data

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -42,9 +61,15 @@ def query_restful():
 
 @click.command()
 @click.option('--mode', '-m', type=click.Choice(['index', 'query']), default='query')
-def main(mode):
-    if mode == 'index':
+# @click.option('--top_k', '-k', default=TOP_K)
```
Comment (nan-wang) [inline app.py:64]:
```suggestion
```

File: app.py [hunk 2]
```diff
@@ -21,6 +22,24 @@ def index():
...
+            fid = str(os.path.splitext(os.path.split(path)[-1])[0])
+            doc = Document(id=fid, uri=path)
+            docs.append(doc)
+        resp = f.post(on='/index', inputs=docs, return_results=True)
+        f.post(on='/dump', parameters={'dump_path': './workspace/dump_lmdb', 'shards': 1})
+        print(f'docs: {len(docs)}')
+        for d in resp[0].docs:
+            print(f'chunks: {len(d.chunks)}')
+            for c in d.chunks:
+                print(f'{c.id}: {c.text}')
```
Comment (nan-wang) [inline app.py:40]:
```suggestion

```

File: app.py [hunk 3]
```diff
@@ -21,6 +22,24 @@ def index():
                 print(f'{c.id}: {c.text}')
 
 
+def index_self_data(data_dir):
+    f = Flow.load_config('flows/index.yml')
+    with f:
+        docs = DocumentArray()
+        for fn in glob.glob(os.path.join(data_dir, '*.vtt')):
+            path = fn
+            fid = str(os.path.splitext(os.path.split(path)[-1])[0])
```
Comment (nan-wang) [inline app.py:31]:
```suggestion
            fid = os.path.splitext(os.path.basename(fn))[0]
```

File: app.py [hunk 4]
```diff
@@ -21,6 +22,24 @@ def index():
                 print(f'{c.id}: {c.text}')
 
 
+def index_self_data(data_dir):
+    f = Flow.load_config('flows/index.yml')
+    with f:
+        docs = DocumentArray()
+        for fn in glob.glob(os.path.join(data_dir, '*.vtt')):
+            path = fn
```
Comment (nan-wang) [inline app.py:30]:
```suggestion
```

File: app.py [hunk 5]
```diff
@@ -21,6 +22,24 @@ def index():
                 print(f'{c.id}: {c.text}')
 
 
+def index_self_data(data_dir):
```
Comment (numb3r3) [inline app.py:25]: `index()` and `index_self_data` is redundant. 

```suggestion
def index(data_dir):
```

File: app.py [hunk 6]
```diff
@@ -42,9 +61,15 @@ def query_restful():
...
-    if mode == 'index':
+# @click.option('--top_k', '-k', default=TOP_K)
+@click.option('--dataset', '-d', type=click.Choice(['toy-data', 'self-data'],case_sensitive=False),
+              default='toy-data')
+@click.option('--data_dir', type=click.Path(exists=True))
+def main(mode, dataset, data_dir):
+    if mode == 'index' and dataset == 'toy-data':
         index()
+    elif mode == 'index' and dataset == 'self-data':
+        index_self_data(data_dir)
```
Comment (numb3r3) [inline app.py:72]:
```suggestion
@click.option('--data_dir', default='toy-data', type=click.Path(exists=True))
def main(mode, data_dir):
    if mode == 'index':
        index(data_dir)
```

**Review by numb3r3 — DISMISSED** [review]
`index()` and `index_self_data` is redundant

**Review by nan-wang — APPROVED** [review]

**Pull request merged**